src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -86,5 +132,53 @@ private Task OnFileWatcher(IFileItem file, object obj)
                 _poll = null;
             }
         }
+
+        private async Task<string> GetLastPollStatusAsync()
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+
+                return await apiHubBlob.DownloadTextAsync();
+            }
+            catch (StorageException exception)
+            {
+                if (exception.RequestInformation == null ||
+                    exception.RequestInformation.HttpStatusCode != 404)
+                {
+                    var errorText = string.Format("Encountered a storage exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, exception.Message);
+                    _trace.Error(errorText);
+                }
+                return null;
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+                return null;
+            }
+        }
+
+        private async Task SetNextPollStatusAsync(string nextPoll)
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+                await apiHubBlob.UploadTextAsync(nextPoll);
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while setting the next poll status to the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+            }
+        }
+
+        private CloudBlockBlob GetBlobReference()
+        {
+            // Path to apiHub blob is:
+            // apihubs/{siteName}/{functionName}/{triggerPath}/{connectionStringSetting}
```

Comment:
Hmm isn't it possible for a function to have multiple triggers? I think I saw the support for it in the portal. I was thinking a function might have multiple input triggers and it would be possible for triggers to use the same path or connection and that's why made it like above to guarantee uniqueness. good point on JSON...